tests/functests/instancetype_test.go

```diff
@@ -50,81 +76,175 @@ var _ = Describe("Common instance types func tests", func() {
 		})
 
 		It("can be created when enough resources are provided", func() {
-			instanceTypeMatcher := v1.InstancetypeMatcher{
-				Name: "u1.large",
-			}
-
-			clusterPreferences, err := virtClient.VirtualMachineClusterPreference().List(context.Background(), metav1.ListOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			Expect(clusterPreferences.Items).ToNot(BeEmpty())
-
-			for _, preference := range clusterPreferences.Items {
-				vm := randomVMWithInstancetype(&instanceTypeMatcher)
-				vm.Spec.Preference = &v1.PreferenceMatcher{
-					Name: preference.Name,
-				}
-
+			for _, preference := range getClusterPreferences(virtClient) {
+				vm = randomVM(&v1.InstancetypeMatcher{Name: "u1.large"}, &v1.PreferenceMatcher{Name: preference.Name}, false)
 				vm, err = virtClient.VirtualMachine(testNamespace).Create(context.Background(), vm)
 				Expect(err).ToNot(HaveOccurred())
-				Eventually(func(g Gomega) {
-					vm, err = virtClient.VirtualMachine(testNamespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
-					g.Expect(err).ToNot(HaveOccurred())
-					g.Expect(vm.Status.Ready).To(BeTrue())
-				}, vmReadyTimeout, time.Second).Should(Succeed())
-
-				err = virtClient.VirtualMachine(testNamespace).Delete(context.Background(), vm.Name, &metav1.DeleteOptions{})
-				Expect(err).ToNot(HaveOccurred())
 			}
 		})
 	})
+
+	Context("VirtualMachine using a preference is able to boot", func() {
+		DescribeTable("a Linux guest with", func(containerDisk, preference, username string, guestAgent bool) {
+			vm = randomVM(&v1.InstancetypeMatcher{Name: "u1.small"}, &v1.PreferenceMatcher{Name: preference}, true)
+			addContainerDisk(vm, containerDisk)
+			privKey := addCloudInitWithAuthorizedKey(vm)
+			vm, err = virtClient.VirtualMachine(testNamespace).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+			expectVMToBeReady(virtClient, vm.Name)
+			if guestAgent {
+				expectGuestAgentToBeConnected(virtClient, vm.Name)
+			}
+			expectSSHToRunCommandWithPrivKey(virtClient, vm.Name, username, privKey)
+		},
+			Entry("Fedora", fedoraContainerDisk, "fedora", "fedora", true),
+			Entry("CentOS 7", centos7ContainerDisk, "centos.7", "centos", true),
+			Entry("CentOS Stream 8", centosStream8ContainerDisk, "centos.stream8", "centos", true),
+			Entry("CentOS Stream 9", centosStream9ContainerDisk, "centos.stream9", "cloud-user", true),
+			Entry("Ubuntu 18.04", ubuntu1804ContainerDisk, "ubuntu", "ubuntu", false),
+			Entry("Ubuntu 20.04", ubuntu2004ContainerDisk, "ubuntu", "ubuntu", false),
+			Entry("Ubuntu 22.04", ubuntu2204ContainerDisk, "ubuntu", "ubuntu", false),
```

Comment:
What is the reason of testing multiple Ubuntu versions? It has costs of time duration during execution of the tests.

Reply:
IMO it makes sense to test all long term releases, as long as they are supported.